src/Symfony/Component/Console/Helper/ProgressIndicator.php

```diff
@@ -0,0 +1,292 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Helper;
+
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ */
+class ProgressIndicator
+{
+    private $output;
+    private $startTime;
+    private $format;
+    private $message = null;
+    private $indicatorValues = array('-', '\\', '|', '/');
+    private $indicatorCurrent = 0;
+    private $indicatorChangeInterval = 100;
+    private $indicatorUpdateTime;
+    private $lastMessagesLength = 0;
+
+    private static $formatters;
+    private static $formats;
+
+    /**
+     * @param OutputInterface $output
+     */
+    public function __construct(OutputInterface $output)
+    {
+        $this->output = $output;
+        $this->startTime = time();
+
+        $this->setFormat($this->determineBestFormat());
+    }
+
+    /**
+     * @param string $format
+     */
+    public function setFormat($format)
+    {
+        $this->format = self::getFormatDefinition($format);
+    }
+
+    /**
+     * @param int $milliseconds
+     */
+    public function setIndicatorChangeInterval($milliseconds)
+    {
+        $this->indicatorChangeInterval = $milliseconds;
+    }
+
+    /**
+     * @param string|null $message
+     */
+    public function setMessage($message)
+    {
+        $this->message = $message;
+
+        $this->display();
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getMessage()
+    {
+        return $this->message;
+    }
+
+    /**
+     * @param array $values
+     */
+    public function setIndicatorValues(array $values)
+    {
+        $values = array_values($values);
+
+        if (empty($values)) {
+            throw new \InvalidArgumentException('Must have at least 1 value.');
+        }
+
+        $this->indicatorValues = $values;
+    }
+
+    /**
+     * Gets the progress bar start time.
+     *
+     * @return int The progress bar start time
+     */
+    public function getStartTime()
+    {
+        return $this->startTime;
```

Comment:
Start time can probably be removed altogether as it's not used anywhere. It's useful in the progress bar as we display time information, but as this is not the case here, I don't see how it can be useful. Or would it be useful to add the possibility to add the elapsed time in the format?